src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -590,21 +632,60 @@ static bool M((string, long) t)
 }";
             var comp = CompileAndVerify(source);
             comp.VerifyDiagnostics();
+
             var tree = comp.Compilation.SyntaxTrees.First();
             var model = comp.Compilation.GetSemanticModel(tree);
 
-            // PROTOTYPE(tuple-equality) Semantic model
-            return;
+            var tuple1 = tree.GetCompilationUnitRoot().DescendantNodes().OfType<TupleExpressionSyntax>().ElementAt(0);
+            var symbol1 = model.GetTypeInfo(tuple1);
+            Assert.Null(symbol1.Type);
+            Assert.Equal("(System.String, System.Int64)", symbol1.ConvertedType.ToTestDisplayString());
+            Assert.Equal("(System.String, System.Int64)", model.GetDeclaredSymbol(tuple1).ToTestDisplayString());
```

Comment:
>model.GetDeclaredSymbol(tuple1) [](start = 58, length = 31)

Interesting, does this API work this way on tuple literals in other context?

Reply:
I think so, and that is what I expected. A tuple literal is where a tuple and its elements are declared. When in the IDE, `tuple.Alice` highlights the element.
I'll add a test to confirm.

---
In reply to: [173336165](https://github.com/dotnet/roslyn/pull/25297#discussion_r173336165) [](ancestors = 173336165)